### PR TITLE
[Disability Benefits] Implement month/year-only dates on toxic exposure pages

### LINKED ISSUE #128641: Replace logic for Toxic Exposure date to accept no/partial date
### User Need

As a frontend engineer
I want to ensure date components allow partial or no dates on Toxic Exposure dates
So that the veteran/user can successfully submit their claim

### Description

Lighthouse accepts month-year, year only, or no date. Conditions V2 workflow logic allows no or partial date by leveraging the Lighthouse API’s affordances.  We will use Conditions V2 logic for a cleaner solution by maintaining consistency of logic.

Based on @freeheeling 's recommendation in https://github.com/department-of-veterans-affairs/va.gov-team/issues/125144#issuecomment-3560695835 and [OCTO's response to the recommendation](https://dsva.slack.com/archives/C0983077SVA/p1766511005148799?thread_ts=1766505485.566059&cid=C0983077SVA)

### Tasks

- [ ] 

### Acceptance Criteria

_No response_

### Engineering

- [ ] All tests pass
- [ ] New functionality is covered by unit tests
- [ ] Logging and monitoring are implemented (if applicable)

### Code Review & Pull Requests

- [ ] PR includes Local testing steps
- [ ] PR includes Flipper/testing state details (if applicable)
- [ ] PR includes Author's local proof of submission screenshot
- [ ] Copilot review completed and feedback addressed
- [ ] Internal reviewer approved
- [ ] Internal reviewer added local proof of submission screenshot
- [ ] Code functionality verified on Staging after merge

### PR DESCRIPTION
**Pull request by freeheeling**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
  - Migrated toxic exposure date inputs from full dates (day/month/year) to month/year-only (e.g., "June 1988" or "1988")
  - Added backward compatibility to accept existing full dates (YYYY-MM-DD) while validating only year and month
  - Updated UI components from `va-memorable-date` to `va-date` to support month/year-only input
- _(If bug, how to reproduce)_
  - _n/a_
- _(What is the solution, why is this the solution)_
  - Veterans should have the option to provide no date, a partial date (year only), only a start/end date, or select that they are not sure of the dates
  - The partial date format also better resembles the data transferred onto the corresponding PDF, which doesn't include a day field 
- _(Which team do you work for, does your team own the maintenance of this component?)_
  - 🌊 Disability Benefits Pathways team
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
  - _n/a_

## Related issue(s)

- closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/128641

## Testing done

- _Describe what the old behavior was prior to the change_
  - Previously, toxic exposure pages required full dates (day/month/year) using the `va-memorable-date` component, and validation enforced complete dates. The pages now accept month/year-only inputs (e.g., "June 1988" or "1988") using `va-date`, with backward compatibility for existing full dates.
- _Describe the steps required to verify your changes are working as expected_
  1. Access toxic exposure pages: Log in and navigate to the disability benefits all-claims form. Select toxic exposure locations to reach the detail pages.
  2. Test month/year-only input: On a detail page, enter dates using only month and year. Verify the form accepts these without requiring a day.
  3. Test year-only input: Enter a year-only date. Verify it's accepted.
  4. Test validation rules:
    - Enter a future date and confirm it shows an error.
    - Enter a end date that occurs earlier than a start date and confirm it shows an error.
    - Enter a year before 1900 and confirm it shows an error.
    - Enter an invalid character as part of a year and confirm it shows an error.
    - For Gulf War 1990: Enter an end date before August 1990 and confirm it shows an error; August 1990 or later should be accepted.
    - For Gulf War 2001: Enter an end date before September 2001 and confirm it shows an error; September 2001 or later should be accepted.
  5. Test "I'm not sure" checkbox: Check the "I'm not sure of the dates" checkbox and verify the form submits without requiring dates.
  6. Test omitting start/end date, or not including any dates: Verify the form proceeds to the next page without errors.
  7. Test valid dates: Enter valid month/year dates and confirm the form proceeds to the next page without errors.
  8. Verify summary display: On the summary page, confirm dates display correctly in month/year format (e.g., "June 1988" or "1988") or as "No dates entered".
  9. Test with a user who already has a full date (day/month/year) stored, that the data are compatible with the new date format.
- _Describe the tests completed and the results_
  - Updated unit tests for 6 toxic exposure detail pages to verify `va-datef rendering, month/year-only formats (YYYY-MM, YYYY-XX), backward compatibility with full dates (YYYY-MM-DD), and the "I'm not sure of the dates" checkbox
  - Added unit tests for custom validation functions covering Gulf War 1990/2001 date ranges, future date checks, optional dates, and edge cases (e.g., August 1990, September 2001)
   - Updated `ToxicExposureSummary` component tests to verify date display formats and "not sure" checkbox behavior
  - All tests passing; CI checks resolved (duplicate property names, schema validation)

## Screenshots

| | Before | After |
| - | ------ | ----- |
| Gulf War 1990 | Screenshot 2026-01-12 at 2 31 12 PM | Screenshot 2026-01-12 at 2 35 10 PM |
| Gulf War 2001 | Screenshot 2026-01-12 at 2 43 02 PM | Screenshot 2026-01-12 at 2 43 57 PM |
| Herbicide with missing year error | Screenshot 2026-01-12 at 2 48 36 PM | Screenshot 2026-01-12 at 2 49 42 PM |
| Additional exposure with invalid date range | Screenshot 2026-01-12 at 2 55 01 PM | Screenshot 2026-01-12 at 2 55 51 PM |
| Future date error | Screenshot 2026-01-12 at 3 02 34 PM | Screenshot 2026-01-12 at 3 01 25 PM |
| Too far past/future error | Screenshot 2026-01-12 at 3 05 31 PM | Screenshot 2026-01-12 at 3 06 45 PM |
| Invalid date input | Screenshot 2026-01-12 at 3 21 35 PM | Screenshot 2026-01-12 at 3 20 23 PM |
| Summary with year only | Screenshot 2026-01-12 at 2 40 08 PM | Screenshot 2026-01-12 at 2 40 38 PM | 
| Summary with no dates entered | _not possible_ | Screenshot 2026-01-12 at 2 45 42 PM |
| Summary with no start date entered | _not possible_ | Screenshot 2026-01-12 at 3 11 31 PM |
| Summary with no end date entered | _not possible_ | Screenshot 2026-01-12 at 3 14 03 PM |
| Summary with not sure only entered | _not possible_ | Screenshot 2026-01-12 at 2 59 21 PM |

## What areas of the site does it impact?

Form 526EZ, Conditions chapter, Toxic Exposure section

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/utils.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+// which is populated after all field blur events
+const getCurrentYear = () => new Date().getFullYear();
+
+/**
+ * Schema pattern that accepts YYYY-MM, YYYY-XX, or YYYY-MM-DD for backward compatibility
+ * This extends the platform's currentOrPastMonthYearDateSchema to also accept full dates
+ */
+export const monthYearDateSchemaWithFullDateSupport = {
+  type: 'string',
+  pattern: '^(\\d{4}|XXXX)-(0[1-9]|1[0-2]|XX)(-\\d{2})?$',
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/utils.js]: Addressed in [c4526e3](https://github.com/department-of-veterans-affairs/vets-website/pull/41319/commits/c4526e3c989963a0f7d198fc0a49e28be4e4d81d)

File: src/applications/disability-benefits/all-claims/utils/dates/validations.js [hunk 2]
```diff
@@ -282,3 +282,68 @@ export const validateApproximateDate = (errors, dateString, options = {}) => {
...
+  }
+
+  // Validate month if present
+  if (isMonthValid) {
+    const m = Number(month);
+    if (!Number.isInteger(m) || m < 1 || m > 12) {
+      errors.addError('Please enter a valid month');
+    }
+  }
+};
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/utils/dates/validations.js]: Addressed in [c4526e3](https://github.com/department-of-veterans-affairs/vets-website/pull/41319/commits/c4526e3c989963a0f7d198fc0a49e28be4e4d81d)

File: src/applications/disability-benefits/all-claims/utils/dates/validations.js [hunk 3]
```diff
@@ -282,3 +282,71 @@ export const validateApproximateDate = (errors, dateString, options = {}) => {
...
+
+  // Parse year as number if it exists and is not XXXX
+  const yearNum = year && year !== 'XXXX' && year !== '' ? Number(year) : null;
+
+  // If year is present but not a valid integer, show generic error
+  if (yearNum !== null && !Number.isInteger(yearNum)) {
+    errors.addError('Please enter a valid date');
+    return;
+  }
+
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/utils/dates/validations.js]: The concern is valid but low priority. The schema prevents this case, and the generic message is acceptable.

File: src/applications/disability-benefits/all-claims/utils/dates/formatting.js [hunk 4]
```diff
@@ -128,12 +128,49 @@ export const formatDateRange = (dateRange = {}, formatStr = DATE_FORMAT) => {
...
+    const [year, month] = rawDate.split('-');
+    const monthNum = parseInt(month, 10);
+    if (monthNum >= 1 && monthNum <= 12) {
+      const date = parse(`${year}-${month}-01`, 'yyyy-MM-dd', new Date());
+      if (isValid(date)) {
+        return format(date, 'MMMM yyyy');
+      }
+    }
+    return '';
+  }
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/utils/dates/formatting.js]: The comment documents the behavior, and the function handles these formats gracefully. The current approach provides backward compatibility without adding noise.

**Review by ZachMurray — APPROVED** [review]
image

LGTM - reviewed, tested, and ran locally well. 

image

**Comment by LinnJS** [comment]
Tested with the following combined work

[rel: vets-json-schema latest version (25.2.43) bump #26085](https://github.com/department-of-veterans-affairs/vets-api/pull/26085)
[rel: vets-json-schema latest version (25.2.43) bump #41640](https://github.com/department-of-veterans-affairs/vets-website/pull/41640)

Submission went through successfully and shape was as expected.

Screenshot 2026-01-22 at 12 01 32 PM

**Review by LinnJS — APPROVED** [review]
